schema/Metrics.v1.0.0.schema.json

```diff
@@ -0,0 +1,354 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "description": "A Metric object that describes the metric.",
+  "required": [
+    "id",
+    "lifecycle",
+    "displayName",
+    "description",
+    "tags",
+    "desiredDirection",
+    "definition"
+  ],
+  "properties": {
+    "id": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 50,
+      "pattern": "^[a-z_][a-z0-9_]*$",
+      "description": "An ID used to uniquely identify and reference the metric.",
+      "readOnly": true
+    },
+    "lifecycle": {
+      "description": "The stage in the metric lifecycle, which determines when the metric is calculated.",
+      "type": "string",
+      "enum": ["Active", "Inactive"],
+      "x-ms-enum": {
+        "name": "LifecycleStage",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Active",
+            "value": "Active",
+            "description": "Metric is included in experiment analysis"
+          },
+          {
+            "name": "Inactive",
+            "value": "Inactive",
+            "description": "Metric is not included in experiment analysis"
+          }
+        ]
+      }
+    },
+    "displayName": {
+      "type": "string",
+      "description": "A display name for the metric to use for display rather than the ID.",
+      "minLength": 1,
+      "maxLength": 200
+    },
+    "description": {
+      "type": "string",
+      "description": "A description of the metric.",
+      "minLength": 1,
+      "maxLength": 1000
+    },
+    "tags": {
+      "type": "array",
+      "description": "The tags of the metric.",
+      "items": {
+        "type": "string",
+        "description": "A tag attached to the metric.",
+        "minLength": 1,
+        "maxLength": 30,
+        "pattern": "^[a-zA-Z0-9 _:()&-]+$"
+      }
+    },
+    "desiredDirection": {
+      "description": "Whether an increase or decrease to the metric value is desired.",
+      "type": "string",
+      "enum": ["Increase", "Decrease", "Neutral"],
+      "x-ms-enum": {
+        "name": "DesiredDirection",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Increase",
+            "value": "Increase",
+            "description": "An increase to the metric value is desired. E.g., Success rate, Total revenue."
+          },
+          {
+            "name": "Decrease",
+            "value": "Decrease",
+            "description": "A decrease to the metric value is desired. E.g., Error rate, Latency."
+          },
+          {
+            "name": "Neutral",
+            "value": "Neutral",
+            "description": "Neither an increase nor a decrease to the metric value is desired, or the desired direction depends on other factors. E.g., Number of video play attempts, Number of user feedbacks"
```

Comment:
Typo nit
```suggestion
            "description": "Neither an increase nor a decrease to the metric value is desired, or the desired direction depends on other factors. E.g., Number of video play attempts, Number of user feedbacks."
```

Reply:
Removed since x-ms-enum is not part of standard draft-07 schema